### PR TITLE
Some more index stuff

### PR DESCRIPTION
**Pull request by annando**
Some more indexes are removed, added or changed. And the text variable types are changed to more reasonable values.

Additionally there is now a mechanism to log index usage.

And finally the dbstructure update mechanism now only uses the old update mechanism when an unique index is added. There is also a (non working) preparation for newer MySQL versions that have problems with the "IGNORE" keyword (which we have to improve to make it work).

### CODE REVIEW

**Comment by annando** [comment]
@rabuzarus @Hypolite @Quix0r do you have some time for a review? I added several commits after I created that pull request but now I think it is stable and running.

File: include/dba.php [hunk 1]
```diff
@@ -138,6 +138,57 @@ public function num_rows() {
 		return $return;
 	}
 
+	public function log_index($query) {
```
Comment (rabuzarus) [inline include/dba.php]: here should be some doxygen header

File: include/poller.php [hunk 2]
```diff
@@ -153,8 +154,24 @@ function poller_execute($queue) {
 	$funcname = str_replace(".php", "", basename($argv[0]))."_run";
 
 	if (function_exists($funcname)) {
+
 		logger("Process ".$mypid." - Prio ".$queue["priority"]." - ID ".$queue["id"].": ".$funcname." ".$queue["parameter"]);
 
+		$stamp = (float)microtime(true);
+
+		if (Config::get("system", "profiler")) {
```
Comment (rabuzarus) [inline include/poller.php]: a comment for this block wouldn't be bad. something like `// Initialize the profiler`

File: include/poller.php [hunk 3]
```diff
@@ -164,13 +181,52 @@ function poller_execute($queue) {
 
 		$a->process_id = $old_process_id;
 
+		$duration = number_format(microtime(true) - $stamp, 3);
+
+		logger("Process ".$mypid." - Prio ".$queue["priority"]." - ID ".$queue["id"].": ".$funcname." - done in ".$duration." seconds.");
+
+		if (Config::get("system", "profiler")) {
```
Comment (rabuzarus) [inline include/poller.php]: here could also be a short comment which describes was the block in the condition does

File: mod/ping.php [hunk 4]
```diff
@@ -195,13 +196,20 @@ function ping_init(App $a)
...
 
-		$ev = qu("SELECT count(`event`.`id`) AS total, type, start, adjust FROM `event`
-			WHERE `event`.`uid` = %d AND `start` < '%s' AND `finish` > '%s' and `ignore` = 0
-			ORDER BY `start` ASC ",
-			intval(local_user()),
-			dbesc(datetime_convert('UTC', 'UTC', 'now + 7 days')),
-			dbesc(datetime_convert('UTC', 'UTC', 'now'))
-		);
+		$cachekey = "ping_init:".local_user();
+		$ev = Cache::get($cachekey);
```
Comment (rabuzarus) [inline mod/ping.php]: Some note: At the present time we don't make use of the event notifier in any of our themes. So I'm fine with this change. But in general it doesn't seem to me a good solution to cache stuff over an hour where the user would expect up-to-date data.

Comment (annando) [inline mod/ping.php]: It is no problem to only update this value once in a while. The system tells about upcoming events up to seven days in the future. On hour less or more really doesn't matter.

**Comment by annando** [comment]
@Hypolite I will mark this day in my calendar. Either I really worked well - or you simply are suffering from a giant jet lag.

File: database.sql [hunk 5]
```diff
@@ -638,7 +635,7 @@ CREATE TABLE IF NOT EXISTS `manage` (
 	`uid` int(11) NOT NULL DEFAULT 0,
 	`mid` int(11) NOT NULL DEFAULT 0,
 	 PRIMARY KEY(`id`),
-	 INDEX `uid_mid` (`uid`,`mid`)
+	 UNIQUE INDEX `uid_mid` (`uid`,`mid`)
```
Comment (Quix0r) [inline database.sql]: Have you validated that the combination of `uid` and `mid` can be really unique?

File: mod/network.php [hunk 6]
```diff
@@ -762,24 +762,25 @@ function network_content(App $a, $update = 0) {
...
+		if (dbm::is_result($unseen)) {
 			$r = q("UPDATE `item` SET `unseen` = 0
 				WHERE `unseen` = 1 AND `uid` = %d",
 				intval(local_user())
 			);
-	}
-	else {
-		if($update_unseen) {
+		}
+	} else {
```
Comment (Quix0r) [inline mod/network.php]: Can these `else` and `if` be combined into `elseif` ?

Comment (Hypolite) [inline mod/network.php]: Absolutely, good eye! 👍

File: database.sql (hunk 5 above)
Comment (annando) [inline database.sql]: I guess so.

**Comment by annando** [comment]
I just added two bugfixes: 

1. We always had stored config values although the same value was already stored.
2. The storing of -1 for the table optimize hadn't worked.

File: include/poller.php [hunk 7]
```diff
@@ -180,14 +162,111 @@ function poller_execute($queue) {
 }
 
 /**
+ * @brief Execute a function from the queue
+ *
+ * @param array $queue Workerqueue entry
+ * @param string $funcname name of the function
+ */
```
Comment (rabuzarus) [inline include/poller.php]: `@param string $argv` is missing

**Comment by Quix0r** [comment]
I think I check to lazy ... :-/

File: include/poller.php (hunk 7 above)
Comment (Quix0r) [inline include/poller.php]: I think I check to lazy ... :-(

Comment (annando) [inline include/poller.php]: Done

File: database.sql (hunk 5 above)
Comment (Quix0r) [inline database.sql]: I asked so because it already happened to me that I didn't validate it and then had a lot trouble.

**Pull request merged**